### PR TITLE
cleanup resolver

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: packages/ember-application/lib/system/resolver.js [hunk 1]
```diff
@@ -117,11 +117,9 @@ export default EmberObject.extend({
   init() {
     this._parseNameCache = dictionary(null);
   },
+
   normalize(fullName) {
-    let [
-      type,
-      name
-    ] = fullName.split(':', 2);
```
Comment (bekzod) [inline packages/ember-application/lib/system/resolver.js]: last time I check second arguments slows down split, (I think thats why its not widely used in code)

File: packages/ember-application/lib/system/resolver.js [hunk 2]
```diff
@@ -117,11 +117,9 @@ export default EmberObject.extend({
...
+    let [ type, name ] = fullName.split(':');
```
Comment (stefanpenner) [inline packages/ember-application/lib/system/resolver.js]: also split `[ type, name ]` (unless loose mode is enabled for babel) produced some gnarly output.

If we are not using loose, lets move away from array destructuring, if we are we can leave it.

**Review by stefanpenner — CHANGES_REQUESTED** [review]

File: packages/ember-application/lib/system/resolver.js (hunk 2 above)
Comment (bekzod) [inline packages/ember-application/lib/system/resolver.js]: I think it generates alright code https://cl.ly/2X0E1Z0Z163s/Screen%20Shot%202017-07-14%20at%209.46.10%20AM.jp2

Comment (stefanpenner) [inline packages/ember-application/lib/system/resolver.js]: looks great, looks like we are correctly using loose mode. Thanks for checking!

**Review by stefanpenner — APPROVED** [review]

**Comment by stefanpenner** [comment]
restarting sauce..

**Pull request merged**